signingscript/script.py

```diff
@@ -40,8 +41,14 @@
                 raise Exception("GPG format is enabled but gpg_pubkey is not defined")
             if not os.path.exists(context.config['gpg_pubkey']):
                 raise Exception("gpg_pubkey ({}) doesn't exist!".format(context.config['gpg_pubkey']))
-        log.info("getting token")
-        await get_token(context, os.path.join(work_dir, 'token'), cert_type, all_signing_formats)
+
+        if 'mar' in all_signing_formats and context.config.get('enable_autograph', False):
+            _ = get_autograph_url_and_auth(context.config)
+
+        if context.config.get('require_signingserver_tokens', True):
+            log.info("getting token")
+            await get_token(context, os.path.join(work_dir, 'token'), cert_type, all_signing_formats)
```

Comment:
As mentioned earlier, we may want to key off the server config here and request a token if there's a requested format that points at a signing server, and get the autograph url/auth if there's a requested format that points at autograph.